visgraph/graphcore.py

```diff
@@ -95,6 +97,25 @@ def fromJson(self, graph):
             node2 = (n2, graph['nodes'][n2])
             self.addEdge(node1, node2, eid=eid, eprops=eprops)
 
+    def clone(self, graph):
+        '''
+        duplicate another graph's contents.  subclasses may wish to modify this
+        function to duplicate more properties if present.
+        '''
+        self.wipeGraph()
+
```

Comment:
An API which directly accesses these internal data structures will never be used to synchronize graphs which live on two different hosts via cobra.  To clone a graph in a way that's compatible with distributed processing ideals, there would need to be a "primitives only" serialization format and emit/parse type APIs with firm boundaries.